### PR TITLE
feat: rename config files to be TypeScript files

### PR DESCRIPTION
**Pull request by thetutlage**
Relies on the latest RC release. Therefore, do not merge it until the RC release is out.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,17 +1,17 @@
 {
   "compilerOptions": {
     "target": "ES2021",
-    "allowJs": true,
```
Comment (thetutlage) [inline tsconfig.json:4]: No need to include JS files by default, since we will have all the source in TypeScript

File: tsconfig.json [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
     "esModuleInterop": true,
     "module": "Node16",
     "moduleResolution": "Node16",
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
     "skipLibCheck": true,
     "skipDefaultLibCheck": true,
-    "declaration": true,
```
Comment (thetutlage) [inline tsconfig.json:12]: The `declaration` files are helpful for libraries as they provide intellisense for compiled code. However, with an application they are not super useful.

**Review by olivermrbl — APPROVED** [review]
LGTM

**Comment by olivermrbl** [comment]
@thetutlage, it seems the tests are broken. Could you look into that? 🙏

**Comment by thetutlage** [comment]
> @thetutlage, it seems the tests are broken. Could you look into that? 🙏

Yeah, they were broken because of not using the latest versions. Now I have updated the packages and it should be all good!

**Pull request merged**